src/CommunityVoices/App/Api/Component/Controller.php

```diff
@@ -24,20 +24,25 @@ protected function getId($request)
     }
 
     // Takes a string list of form attributes and returns an array of them from the request.
-    protected function getFormAttributes($request, $attributes, $useDefaults = true)
+    protected function getFormAttributes($request, $attributes, $useDefaults = true, $identifier = null)
```

Comment:
This feels a little bit too much like forcing this controller-wide function into working for batch upload ... I guess it's okay, but I just wonder about `postBatchUpload` - the only function that uses this adjustment - writing out its differences only there.  Not necessary to change this, but just of note.

Reply:
@Sammidysam fair point, and I could make a second function if need be.

Reply:
@Sammidysam This part of the function seems entirely unnecessary to be in getFormAttributes(

            return array_filter($requestAttributes, function ($value) {
                return is_array($value) || !is_null($value);
            });

This seems entirely unncessesary to me, and if this section of code is removed, I wouldn't need to treat batch upload any differently. The reason I had to make a fourth param is because this part of the function will strip out any (potential) request attributes that go unused, but this actually ends up being problematic. Can we just get rid of this code controllerWide? I can't think of a real use case for this. This would also allow me to get rid of the extra param.

Reply:
This is needed because of editing.  When editing from a form, we have two cases:
1) API -> only what is to be changed should be passed in the body.
2) Website -> everything on the form gets passed, regardless of if there is change or not.

The issue with this comes with certain form elements, specifically checkboxes.  For the list of potential content categories, I believe it would be non-existent from the form at all when there are no boxes checked even if the request is coming from the website.  If we treat no content categories at all in the request as delete all content categories, then any API request has to pass along the existing content categories in order to not have them deleted, thus breaking the pass-what-changed paradigm.  So, I believe the point of this logic was to be able to distinguish between a website deletion of checkboxes and an API request not modifying checkboxes.  It is rather convoluted and annoying.

I hope that helps with the rationale?  I think having a separate batch upload function that filters this to only what batch upload needs is probably preferred, since I can't see your new argument being used anywhere else.

Let me know if you have any questions - it has been a long while since I dealt with this code block above.